core/shared/src/main/scala/zio/metrics/MetricKey.scala

```diff
@@ -29,8 +29,8 @@ import zio.stacktracer.TracingImplicits.disableAutoTrace
 final case class MetricKey[+Type] private (
   name: String,
   keyType: Type,
-  tags: Set[MetricLabel] = Set.empty
-) { self =>
+  tags: Set[MetricLabel]
+)(val description: Option[String]) { self =>
```

Comment:
The curried parameter here is very weird.

Reply:
Do you prefer custom `equals / hashcode`  implementation? We cannot include description into identity.

Reply:
I don't think that is a problem other than we want to distinguish what is a tag versus what is the help.

Reply:
Tough question. Probably it's a good idea to be consistent with Micrometer approach, which have key identity only on name and tags:
https://github.com/micrometer-metrics/micrometer/blob/4c00160d34d490e2599002a6e8e5d1a7405160ce/micrometer-core/src/main/java/io/micrometer/core/instrument/Meter.java#L358

Reply:
Having description in key identity may also affect key lookup performance